interfaces/hotplug/deviceinfo.go

```diff
@@ -0,0 +1,94 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package hotplug
+
+import (
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+// HotplugDeviceInfo carries information about added/removed device detected at runtime.
+type HotplugDeviceInfo struct {
+	// kobj name reported by uevent (corresponds to DEVPATH).
+	object string
+	// map of all attributes returned for given uevent.
+	data map[string]string
+}
+
+// NewHotplugDeviceInfo creates HotplugDeviceInfo structure related to udev add or remove event.
+func NewHotplugDeviceInfo(obj string, env map[string]string) *HotplugDeviceInfo {
+	return &HotplugDeviceInfo{
+		object: obj,
+		data:   env,
+	}
+}
+
+// Returns object path, i.e. the sysfs path of the device, e.g. /devices/pci0000:00/0000:00:14.0/usb1/1-2.
+// It is expected to be equal to DEVPATH attribute.
+func (h *HotplugDeviceInfo) Object() string {
```

Comment:
I really think this should be just `DevPath` or `Devpath` and that we should not use the un-decorated word `object`.

Reply:
Ok, removed object (and Object getter) alltogether for now, per IRC discussion.